javascript/extractor/src/com/semmle/js/extractor/FileExtractor.java

```diff
@@ -42,7 +42,68 @@
       Pattern.compile("^(?s)\\s*\\{\\s*\"([^\"]|\\\\.)*\"\\s*:.*");
 
   /** The charset for decoding UTF-8 strings. */
-  private static final Charset UTF8_CHARSET = Charset.forName("UTF-8");
+  private static final Charset UTF8_CHARSET = StandardCharsets.UTF_8;
+
+  /**
+   * Returns true if the byte sequence contains invalid UTF-8 or unprintable ASCII characters.
+   */
+  private static boolean hasUnprintableUtf8(byte[] bytes, int length) {
```

Comment:
I trust that this is a verbatim move.

Reply:
> I trust that this is a verbatim move.

It is, byte for byte - except for adjusted indentation.